### PR TITLE
Add good software training

### PR DESCRIPTION
**Pull request by bobturneruk**

### CODE REVIEW

**Review by Robadob — COMMENTED** [review]
Why does the date in the header say `2021-10-07` and the url says `2021-10-01`?

Even is 7th, so perhaps update the URL/filename to match.

File: _events/workshop-2021-10-01-good-soft.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+**If you want the software you make for research to produce trustworthy results and to be reusable by others on different computers, this course will help.**
+
+*This event is for University of Sheffield researchers and research students only. Please register with an **@shef.ac.uk** or **@sheffield.ac.uk** email address to ensure your ticket request is accepted.*
+
+**Prerequisite skills:** Some python programming experience.
+
+**Learning Objectives**
+
+* Add documentation directly into python code
```
Comment (Robadob) [inline _events/workshop-2021-10-01-good-soft.md]: Will you actually be covering any specific python documentation generation tool?

If so, might be better to reword "Commenting functions and classes to generated documentation using x (and more)". 

But I can see, that you're trying to differentiate it from writing a standalone document describing the code.

File: _events/workshop-2021-10-01-good-soft.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+* How and when to make your code open source
+* Write tests so you can update your code with confidence
```
Comment (Robadob) [inline _events/workshop-2021-10-01-good-soft.md]: Again, perhaps mention pytest or whatever you plan on covering. Some people might already have some awareness of it, but have never used it.

File: _events/workshop-2021-10-01-good-soft.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+This is a pilot course and as such we may encounter problems that we wouldn't normally expect. We will use feedback from this event to improve in future!
```
Comment (Robadob) [inline _events/workshop-2021-10-01-good-soft.md]: to improve 'the course' in future! 
to improve future iterations of the course!
?

**Review by Robadob — APPROVED** [review]
If you address any of my comments, probably update the copy on the eventbrite too for consistency.

File: _events/workshop-2021-10-01-good-soft.md (hunk 1 above)
Comment (bobturneruk) [reply _events/workshop-2021-10-01-good-soft.md]: Thanks @Robadob - it's just about docstrings - I'll be more specific.

File: _events/workshop-2021-10-01-good-soft.md (hunk 2 above)
Comment (bobturneruk) [reply _events/workshop-2021-10-01-good-soft.md]: good point, thanks!

File: _events/workshop-2021-10-01-good-soft.md (hunk 3 above)
Comment (bobturneruk) [reply _events/workshop-2021-10-01-good-soft.md]: thanks!

File: _events/workshop-2021-10-01-good-soft.md (hunk 1 above)
Comment (Robadob) [reply _events/workshop-2021-10-01-good-soft.md]: If you're teaching docstrings, might aswell demonstrate one of the tools which can convert it into a fancy apidoc website/thing, even if it's just a brief slide.

Comment (bobturneruk) [reply _events/workshop-2021-10-01-good-soft.md]: Yes - however, being the pessimist I am, I'm stating the minimum we can do based on the current material! (We have from now until the event to adapt the material, but if I waited for it to be to our satisfaction before advertising the course, I'd wait a while!).

Comment (Robadob) [reply _events/workshop-2021-10-01-good-soft.md]: Fair, update it in hindsight/next time then I guess.

Comment (bobturneruk) [reply _events/workshop-2021-10-01-good-soft.md]: Cheers, yes, probably as we go. This is a pilot.

**Pull request merged**